### ISSUE TITLE
indexing

### ISSUE THREAD

**Issue by gedw99** [vjeantet/hugo-theme-docdock#96]
I can add bleve as a search engine if you want.

**Comment by vjeantet** [comment]
Yes can you do that as a optional search engine to the existing one ?

**Comment by gedw99** [comment]
Yes I think optional is possible.

When I read that this projects code exports the full path to each page that
what made me see the opportunity to pump the URL and page data into bleve.

I am assuming I can just send the whole page.index to bleve. But it will
also I decided the navigation which is probably not what we want.

It won't work the same way with bleve.
It will bring up a Google like search page once the user has put their
search criteria into the search box.
Will require bleve to be running on the server too.

So it's not all rises and champagne or whatever the term is

I recall also there is a failed Hugo with bleve project out there

**Comment by vjeantet** [comment]
I think this behaviour is fully acceptable and I already have usecase in mind.

Share your work I will try to contribute

**Comment by gedw99** [comment]
Ok game on.

Give me some time.

**Comment by gedw99** [comment]
Btw i  am really interested in the mermaid diagrams.
I saw a few JavaScript based editors out there as we as react ones too..
Wondering if there is a golang based editor out there or if your ode
provides editting ?
The intent is to allow editting if these diagrams within the Hugo system.

**Comment by vjeantet** [comment]
Hello @gedw99, any update ?
